tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/retry/clientserver/RetryClientWithDelay.java

```diff
@@ -35,30 +36,35 @@
 public class RetryClientWithDelay {
     private int counterForInvokingConnenectionService;
     private long timestampForConnectionService = 0;
-    private Set<Long> delayTimes = new HashSet<Long>();
+    private Set<Duration> delayTimes = new HashSet<>();
+    
+    // Expect delay between 0-800ms. Set limit to 900ms to allow a small buffer
+    private static final Duration MAX_DELAY = Duration.ofMillis(900);
+    
     //There should be 0-800ms (jitter is -400ms - 400ms) delays between each invocation
     //there should be at least 4 retries
     @Retry(delay = 400, maxDuration= 3200, jitter= 400, maxRetries = 10)
     public Connection serviceA() {
         return connectionService();
     }
+    
     //simulate a backend service
     private Connection connectionService() {
-       // the time delay between each invocation should be 0-800ms
-       if (timestampForConnectionService != 0) {
```

Comment:
Note that in the old revision, this condition would never be met because `timestampForConnectionService` would only be set to a non-zero value within this if-block.